### PR TITLE
P2: signup: initial styling

### PR DESCRIPTION
**Pull request by lamosty**
Part of introducing the revamped signup flow for the P2 project.

In this PR, we add the redesigned "site" signup step for P2.

## Testing instructions

Navigate to `http://calypso.localhost:3000/start/p2` and go through the flow. It should look and work fine. Try both in an incognito window (=> new wp.com account) and with your current wp.com account.

### CODE REVIEW

File: client/signup/p2-processing-screen/index.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+					<img
+						src="https://wpcom.files.wordpress.com/2020/06/p2-logo-light-small.png"
+						width="67"
+						height="32"
+						alt="P2 logo"
+					/>
+				</div>
+
+				<h2>Hooray!</h2>
+				<h2>Your new P2 is</h2>
```
Comment (lezama) [inline client/signup/p2-processing-screen/index.jsx:30]: shoulld we translate these?

Comment (lamosty) [reply client/signup/p2-processing-screen/index.jsx:30]: Yes, that's a WIP (as the whole PR) :)

**Comment by lamosty** [comment]
I still need to add translations and unify the css to variables, but it's ready for testing and feedback!

**Comment by lamosty** [comment]
Thank you for the great feedback @griffbrad! Addressed.

File: client/signup/steps/p2-site/index.jsx [hunk 2]
```diff
@@ -253,7 +251,7 @@ class P2Site extends React.Component {
 					className="p2-site__validation-site-title"
 				>
 					<FormLabel htmlFor="site-title-input">
-						{ this.props.translate( "What's the name of your team or project?" ) }
+						{ this.props.translate( 'Name your team or project' ) }
```
Comment (a8ci18n) [inline client/signup/steps/p2-site/index.jsx:254]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Please enter your team or project name.' )` *ES Score: 7*

**Comment by evilluendas** [comment]
I just pushed a few tweaks to the css, mostly overwriting the Calypso styles in the form inputs. I'd like to tweak a couple of things so it looks better in mobile, I'll do it asap and commit the changes.

I've noticed a couple of things that I don't know how to fix:

- When you click on the button it immediately changes to "Creating your site", even if you haven't entered anything. Instead of changing the label, I would disable the button while it's thinking (it'll probably require some tweaks to the CSS of the disabled button too, I'll do them after the change).



- In the gif above you can also see that the first field doesn't get the `is-error` red border when there is an error. I think it's a matter of adding the class.

- In the old-style login signup screen the contents aren't centered but on the left side. I'd say it's this class limiting the width:

image

Aside from that, looks great! I'd like to add some animation to the background of the "Hooray, site is creating" screen but I'd need to pause the flow there instead of automatically go to the frontend and I'll need your help, as we've talked in Slack. 😄

We'll also need to implement [the second step explaining](https://github.com/Automattic/p2/issues/461) that P2 works with wpcom accounts, but we can do it in a separate PR.

**Comment by lamosty** [comment]
Thanks for the styling fixes @evilluendas and sorry for missing those.

I'll look at the rest of the things you've mentioned.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/3857978

Hi @lamosty, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `{{h2}}Hooray!{{br/}}Your new P2 is{{br/}}almost ready.{{/h2}}`
- `Name your team or project`
- `Share, discuss, review, and collaborate across time zones, without interruptions.`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/3857978).

**Pull request merged**